### PR TITLE
Let cosine_distance work across multiple axes

### PR DESCRIPTION
**Pull request by noahl**
This change lets users use `cosine_distance` to directly compute the cosine distance between n-dimensional tensors. Currently this requires reshaping the input tensors to 1D before calling `cosine_distance`. Other loss functions don't require reshaping, so it would be nice to remove this little bit of cognitive overhead.

I also added tests for `cosine_distance`, for both 1- and 2-dimensional cases.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ekelsen: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

File: tensorflow/python/ops/losses/losses_impl.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def cosine_distance(
     predictions.get_shape().assert_is_compatible_with(labels.get_shape())
 
     radial_diffs = math_ops.multiply(predictions, labels)
-    losses = 1 - math_ops.reduce_sum(radial_diffs, axis=(axis,), keepdims=True)
+    losses = 1 - math_ops.reduce_sum(radial_diffs, axis=axis, keepdims=True)
```
Comment (yifeif) [inline tensorflow/python/ops/losses/losses_impl.py]: Thanks for the PR! Could you also update the docstring as well?

**Comment by tensorflowbutler** [comment]
It has been 20 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 49 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
